pkg/cloudprovider/providers/vsphere/vsphere.go

```diff
@@ -681,7 +711,16 @@ func (vs *VSphere) DetachDisk(volPath string, nodeName k8stypes.NodeName) error
 	}
 	requestTime := time.Now()
 	err := detachDiskInternal(volPath, nodeName)
-	vclib.RecordvSphereMetric(vclib.OperationDetachVolume, requestTime, nil)
+	if err != nil {
+		if IsManagedObjectNotFoundError(err) {
```

Comment:
It seems like next 3 lines of code are repeated multiple times. Can you write a separate function for this.

if IsManagedObjectNotFoundError(err) {
glog.V(4).Infof("error %q ManagedObjectNotFound for node %q", err, convertToString(nodeName))
err = vs.nodeManager.RediscoverNode(nodeName)
....